pkg/crc/podman/podman.go

```diff
@@ -0,0 +1,82 @@
+package podman
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	crcos "github.com/code-ready/crc/pkg/os"
+)
+
+const (
+	rootless = "crc"
+	rootful  = "crc-root"
+)
+
+type Config struct {
+	PodmanExecutablePath string
+}
+
+func NewConfig() Config {
+	return Config{
+		PodmanExecutablePath: filepath.Join(constants.CrcOcBinDir, constants.PodmanRemoteExecutableName),
+	}
+}
+
+func (podman Config) runCommand(args ...string) (string, string, error) {
+	return crcos.RunWithDefaultLocale(podman.PodmanExecutablePath, args...)
+}
+
+func (podman Config) addSystemConnection(defaultCon bool, name, identity, uri string) error {
+	// podman system connection add --identity /Users/prkumar/.ssh/podman-machine-default crc ssh://core@localhost:2222/run/user/1000/podman/podman.sock
+	args := []string{"system", "connection", "add", "--identity", identity, name, uri}
+	if defaultCon {
+		args = append(args, "-d")
+	}
+	if _, stderr, err := podman.runCommand(args...); err != nil {
+		return fmt.Errorf("failed to add podman system connection %v: %s", err, stderr)
+	}
+	return nil
+}
+
+// AddRootlessSystemConnection and makes it default
+func (podman Config) AddRootlessSystemConnection(identity, uri string) error {
```

Comment:
maybe rename `identity` to `sshIdentity` over this file to make it more obvious what `identity` is?